src/ArgParse.jl

```diff
@@ -1,10 +1,9 @@
-isdefined(:OptionsMod) || require("Options")
+isdefined(:OptionsMod) || @eval import Options
```

Comment:
Hi,

Thanks for considering this pull request.
For the @eval part, I'm copying the suggestion in
https://github.com/JuliaLang/julia/issues/12233
This change gets rid of the warning about `require` being deprecated.
However the warning "Warning: requiring "Options" did not define a
corresponding module." remains.
Is Options still necessary? The README for that package states that much of
its functionality is now in base.

The removal of Compat.jl was unrelated. Sorry, that should have been a
separate commit at least. It looks like it is no longer necessary? I was
hoping that removing a dependency would get the load time down a bit to
make command line programs a bit snappier. BTW, I was also trying out the
new precompilation feature to get a really fast startup of this package,
but something about that makes julia fail to find Options.OptionsMod.

Regards,

Pete

____________________
Peter M. Haverty, Ph.D.
Genentech, Inc.
phaverty@gene.com

On Tue, Jul 21, 2015 at 5:04 AM, Carlo Baldassi <notifications@github.com>
wrote:

> In src/ArgParse.jl
> <https://github.com/carlobaldassi/ArgParse.jl/pull/14#discussion_r35094490>
> :
>
> > @@ -1,10 +1,9 @@
> > -isdefined(:OptionsMod) || require("Options")
> > +isdefined(:OptionsMod) || @eval import Options
>
> I don't think @eval is needed here, is it?
> Also, I'm not on latest julia master: does this still print a warning?
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/carlobaldassi/ArgParse.jl/pull/14/files#r35094490>.
>